### PR TITLE
 task10

### PR DESCRIPTION
**Pull request by dariadibal**
Unit tests are added. Please review

### CODE REVIEW

File: src/test/java/parser/JsonParserTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package parser;
+
+import org.junit.jupiter.api.*;
+import shop.Cart;
+
+import java.io.File;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class JsonParserTest {
```
Comment (KirillBugrim) [inline src/test/java/parser/JsonParserTest.java:10]: please use access modifiers in class, fields and methods

File: src/test/java/parser/JsonParserTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    static File file;
+    @BeforeAll
+    static void fileCreation(){
```
Comment (KirillBugrim) [inline src/test/java/parser/JsonParserTest.java:13]: please add 1 empty line between lines of code

File: src/test/java/parser/JsonParserTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        file = new File("src/main/resources/raccoon-cart.json");
+    }
+
+    @Tag("happy")
+    @Test
```
Comment (KirillBugrim) [inline src/test/java/parser/JsonParserTest.java:18]: please add description to each test

File: src/test/java/parser/JsonParserTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    void readFromFile() {
+        JsonParser parser = new JsonParser();
+        Cart cart = parser.readFromFile((new File("src/main/resources/eugen-cart.json")));
+
+        assertAll(
+                "group assertion for JsonParser read",
+                ()->assertNotNull(cart),
+                ()->assertEquals(cart.getClass(), Cart.class),
+                ()->assertEquals("eugen-cart", cart.getCartName())
```
Comment (KirillBugrim) [inline src/test/java/parser/JsonParserTest.java:27]: please provide a 3rd parameter - error message for eacg assertions

File: src/test/java/shop/CartTest.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        flash = new VirtualItem();
+        flash.setName("Linux");
+        flash.setPrice(15);
+        flash.setSizeOnDisk(25);
+    }
+
+    @Test
+    void realItemAndVirtualItemTotalPriceCheck(){
+       double expectedTotal = 0;
+       double TAX = 0.2;
```
Comment (KirillBugrim) [inline src/test/java/shop/CartTest.java:34]: please follow java code convention in naming of fields. 
And what the magic number this TAX?

File: src/test/java/shop/VirtualItemTest.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
+package shop;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class VirtualItemTest {
+    String expectedMsg = "Class: class shop.VirtualItem; Name: USB; Price: 15.0; Size on disk: 30000.0";
```
Comment (KirillBugrim) [inline src/test/java/shop/VirtualItemTest.java:8]: please make this field final

**Review by KirillBugrim — COMMENTED** [review]
good, but need to update a little bit

File: src/test/java/parser/JsonParserTest.java (hunk 1 above)
Comment (dariadibal) [reply src/test/java/parser/JsonParserTest.java]: Done

File: src/test/java/parser/JsonParserTest.java (hunk 2 above)
Comment (dariadibal) [reply src/test/java/parser/JsonParserTest.java]: Done

File: src/test/java/parser/JsonParserTest.java (hunk 3 above)
Comment (dariadibal) [reply src/test/java/parser/JsonParserTest.java:19]: Done

File: src/test/java/parser/JsonParserTest.java (hunk 4 above)
Comment (dariadibal) [reply src/test/java/parser/JsonParserTest.java]: Done

File: src/test/java/shop/VirtualItemTest.java (hunk 6 above)
Comment (dariadibal) [reply src/test/java/shop/VirtualItemTest.java]: Done

**Review by KirillBugrim — APPROVED** [review]

**Pull request merged**